src/wazuh_modules/vulnerability_scanner/tests/unit/scanOrchestrator_test.cpp

```diff
@@ -10,24 +10,657 @@
+               .kernelRelease = "osdata_kernelRelease"};
+
+    spOsDataCacheMock = std::make_shared<MockOsDataCache>();
+    EXPECT_CALL(*spOsDataCacheMock, getOsData(_)).WillRepeatedly(testing::Return(osData));
+
+    auto spOsOrchestrationMock =
+        std::make_shared<MockAbstractHandler<std::shared_ptr<TScanContext<TrampolineOsDataCache>>>>();
+    EXPECT_CALL(*spOsOrchestrationMock, handleRequest(_)).Times(0);
+
+    auto spPackageInsertOrchestrationMock =
+        std::make_shared<MockAbstractHandler<std::shared_ptr<TScanContext<TrampolineOsDataCache>>>>();
+    EXPECT_CALL(*spPackageInsertOrchestrationMock, handleRequest(_)).Times(0);
+
+    auto spPackageDeleteOrchestrationMock =
+        std::make_shared<MockAbstractHandler<std::shared_ptr<TScanContext<TrampolineOsDataCache>>>>();
+    EXPECT_CALL(*spPackageDeleteOrchestrationMock, handleRequest(_)).Times(0);
+
+    auto spIntegrityClearOrchestrationMock =
+        std::make_shared<MockAbstractHandler<std::shared_ptr<TScanContext<TrampolineOsDataCache>>>>();
+    EXPECT_CALL(*spIntegrityClearOrchestrationMock, handleRequest(_)).Times(1);
+
+    spFactoryOrchestratorMock = std::make_shared<MockFactoryOrchestrator>();
+    EXPECT_CALL(*spFactoryOrchestratorMock, create())
+        .WillOnce(testing::Return(spOsOrchestrationMock))
+        .WillOnce(testing::Return(spPackageInsertOrchestrationMock))
+        .WillOnce(testing::Return(spPackageDeleteOrchestrationMock))
+        .WillOnce(testing::Return(spIntegrityClearOrchestrationMock));
+
+    auto spIndexerConnectorMock = std::make_shared<MockIndexerConnector>();
+
+    auto spDatabaseFeedManagerMock = std::make_shared<MockDatabaseFeedManager>();
+
+    flatbuffers::Parser parser;
+    ASSERT_TRUE(parser.Parse(syscollector_synchronization_SCHEMA));
+    ASSERT_TRUE(parser.Parse(SYNCHRONIZATION_INTEGRITY_CLEAR_MSG.c_str()));
+    uint8_t* buffer = parser.builder_.GetBufferPointer();
+    std::variant<const SyscollectorDeltas::Delta*, const SyscollectorSynchronization::SyncMsg*>
+        syscollectorSynchronization = SyscollectorSynchronization::GetSyncMsg(reinterpret_cast<const char*>(buffer));
+
+    std::shared_ptr<SocketClient<Socket<OSPrimitives, NoHeaderProtocol>, EpollWrapper>> socketClient =
+        std::make_shared<SocketClient<Socket<OSPrimitives, NoHeaderProtocol>, EpollWrapper>>(TEST_PATH);
+    std::shared_mutex mutexScanOrchestrator;
+
+    TScanOrchestrator<TScanContext<TrampolineOsDataCache>,
+                      TrampolineFactoryOrchestrator,
+                      MockAbstractHandler<std::shared_ptr<TScanContext<TrampolineOsDataCache>>>,
+                      MockIndexerConnector,
+                      MockDatabaseFeedManager>
+        scanOrchestrator(spIndexerConnectorMock, spDatabaseFeedManagerMock, socketClient, mutexScanOrchestrator);
+
+    EXPECT_NO_THROW(scanOrchestrator.run(syscollectorSynchronization));
+}
+
+TEST_F(ScanOrchestratorTest, TestRunOrchestrationThrowException)
```

Comment:
This test name is OK ?

Reply:
I changed the name to "TestRunOrchestrationThatGeneratesException". The intention is to test an orchestration that throws an exception since the method run has a code section to catch them.